src/renderer/components/uielements/qrCode/qrCode.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useRef, useState } from 'react'
+
+import * as RD from '@devexperts/remote-data-ts'
+import { delay } from '@xchainjs/xchain-util'
+import { Grid, Spin } from 'antd'
+import * as FP from 'fp-ts/function'
+import * as O from 'fp-ts/Option'
+import QRCode from 'qrcode'
+import * as ReactDom from 'react-dom'
+import { useIntl } from 'react-intl'
+
+import * as Styled from './qrCode.styles'
+
+type Props = {
+  text: O.Option<string>
+  noDataError: string
```

Comment:
Why is `text` an `Option` and not just a required prop? By required, no handling for `noDataError` is needed anymore.